log/log_level.go

```diff
@@ -0,0 +1,44 @@
+package log
+
+import "strings"
+
+// A log severity level
+type LogLevel int
+
+const (
+	Trace LogLevel = iota
+	Debug
+	// Info-level logs indicate normal service lifecycle events, 'valuable' events for audit, etc.
+	Info
+	// Warning-level logs indicate events that are unexpected but tolerable, which may need further investigation
+	Warning
+	// Error-level logs indicate errors that should be investigated
+	Error
+)
+
+// All log levels
+var LogLevels = [...]LogLevel{Trace, Debug, Info, Warning, Error}
+
+func (l LogLevel) String() string {
+	return [...]string{"TRACE", "DEBUG", "INFO", "WARNING", "ERROR"}[l]
```

Comment:
Does Go create a new object every time this `string{...}` is called?

Reply:
Yep. Cargo culted. Moving the array creation out.